lib/core/src/display/navigation/navigator.rs

```diff
@@ -0,0 +1,158 @@
+use crate::prelude::*;
+
+use super::EventHandler;
+use super::Event;
+use super::ZoomEvent;
+use super::PanEvent;
+use crate::system::web::EventListeningResult as Result;
+use crate::display::rendering::Camera;
+use crate::display::rendering::CameraType;
+use crate::display::rendering::DOMContainer;
+
+use nalgebra::Vector3;
+
+// =================
+// === Animation ===
+// =================
+
+/// Struct used to store animation related properties.
+struct Animation {
+    desired_pos  : Vector3<f32>,
+    vel          : Vector3<f32>,
+    desired_rate : f32,
+    time_count   : f32,
+    drag         : f32,
+    spring_coeff : f32,
+    mass         : f32,
+    min_vel      : f32
+}
+
+// =================
+// === Navigator ===
+// =================
+
+/// Navigator enables camera navigation with mouse interactions on the specified
+/// DOM.
+pub struct Navigator {
+    dom    : DOMContainer,
+    events : Rc<EventHandler>,
+    anim   : Animation
+}
+
+impl Navigator {
+    pub fn new(dom:&DOMContainer, camera:&Camera) -> Result<Self> {
+        let events        = EventHandler::new(dom)?;
+        let dom           = dom.clone();
+
+        let desired_pos   = *camera.position();
+        let vel           = Vector3::new(0.0, 0.0, 0.0);
+        let time_count    = 0.0;
+        let desired_rate  = 120.0; // 120fps
```

Comment:
Why 120FPS? I dont have 120 fps monitor. Plus, if you need to add comment `// 120fps` it means that either the variable or API is wrongly named.

Reply:
Should we target 60fps instead?

Reply:
I also renamed `desired_rate` to `desired_fps`